### ISSUE TITLE
Make request body required by default

### ISSUE THREAD

**Issue by neon-dev** [eclipse/microprofile-open-api#349]
As per @EricWittmann's request following a discussion in https://github.com/smallrye/smallrye-open-api/issues/110#issuecomment-493473681 I propose to make the `required` property of `@RequestBody` true by default
> because more often than not you expect the request body and depend on it

For reference, this is the current state:
https://github.com/eclipse/microprofile-open-api/blob/49e31ea4bf6da48f8491327e1df7f4e47febfc3a/api/src/main/java/org/eclipse/microprofile/openapi/annotations/parameters/RequestBody.java#L59

**Comment by arthurdm** [comment]
Sounds reasonable to me (for `2.0`, which is allowing change in behaviour).

**Comment by EricWittmann** [comment]
This is related to the broader discussion about what default values actually mean in all these annotation properties.  My apologies for missing today's hangout to discuss that.  But we should include this issue as part of that discussion when we have it.

**Comment by EricWittmann** [comment]
Unfortunately I think this is rather problematic.  On the face of it I think it seems obvious and sensible.  However, to understand why this is a problem please read this:

https://github.com/eclipse/microprofile-open-api/issues/356

The issue is that the default values for the annotation properties are actually meant to serve as an indicator to the annotation process that the developer didn't include that property in the application of the annotation.  It may still be the case that changing the default value to true is the right thing to do here, but this needs a bit more consideration in light of #356.

**Comment by EricWittmann** [comment]
I think we should discuss this again in the next meeting.

**Comment by neon-dev** [comment]
As you've stated, the underlying problem is that annotation property default values unfortnunately cannot be null. So there's always a problem with non-String Java types where you cannot detect during processing if the user defined the value.
Knowing this, I still think it wouldn't be too bad setting this to true here. But if there are concerns you could consider changing such boolean types to a tri-state enum of `TRUE`, `FALSE` and `UNSET`. Then you can apply the default value logic if it's `UNSET`.

**Comment by EricWittmann** [comment]
That's a reasonable suggestion for booleans, certainly.  Although in this case I think you'd still need to update the javadoc to inform the developer of what is actually going to be generated.  Because if we do this then the default value goes from `false` to `UNSET`.  So far so good, but what you're really asking for in this Issue is for the value of `required` in the generated OpenAPI doc **when the developer omits the required property from the annotation** to be `true`.

So for that to be the case, it seems to me that the only way to inform the developer of that fact is to update the javadoc.

**Comment by neon-dev** [comment]
Yup, a Javadoc update would certainly be necessary then and you correctly understood where I was going with this (if preferring the the enum approach).

**Comment by neon-dev** [comment]
The thing to really consider in that case would be to do this for all current boolean properties, like in `@Schema` and so on. Mainly for consistency reasons. The main con for using an enum would be that it's kind of an overkill. Because effectively there are only two possible values, and the developer will only ever set one of them, namely the non default value.
That's why personally I'd not go the enum route here.

**Comment by EricWittmann** [comment]
After further discussion in today's bi-weekly specification meeting, we have reached a consensus that we will, in fact, change the default value of the `reqired` property to `true`.  This does make sense in the framework of default values of annotations, but it represents an exception relative to other annotation property default values.  An explanation of what I mean by that can be found here:

https://github.com/eclipse/microprofile-open-api/issues/356#issuecomment-502763082

The tldr is that we'll make the following changes:

* Change the default value of this to `true`
* Update the javadoc to explain why it's true when most other boolean property defaults are `false`
* Update the text of the specification to better explain default annotation properties more broadly
* Maintain a table of exceptions (like this one) where appropriate
* Add a TCK test to ensure that implementations properly handle this case

**Comment by HerrDerb** [comment]
anything happend here? I still cluttering my code with `@Schema(required=true)` :)

**Comment by Azquelt** [comment]
Not yet :cry: It's still in the 4.0 milestone and we haven't made much progress with that.

**Comment by HerrDerb** [comment]
@andrius-kurtinaitis I use Quarkus 3.x.x which switched to Jakarta. There the Jakarta @NotNull works. This might be framework specific
